src/model/application/interaction.rs

```diff
@@ -289,11 +293,101 @@ bitflags! {
     }
 }
 
+/// A cleaned up enum for determining the authorizing owner for an [`Interaction`].
+///
+/// [Discord Docs](https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-object-authorizing-integration-owners-object)
+#[cfg_attr(feature = "typesize", derive(typesize::derive::TypeSize))]
+#[cfg(feature = "unstable_discord_api")]
+#[derive(Clone, Debug)]
+#[non_exhaustive]
+pub enum AuthorizingIntegrationOwner {
+    /// The [`Application`] was installed to a guild, containing the id if invoked in said guild.
+    ///
+    /// [`Application`]: super::CurrentApplicationInfo
+    GuildInstall(Option<GuildId>),
+    /// The [`Application`] was installed to a user, containing the id of said user.
+    ///
+    /// [`Application`]: super::CurrentApplicationInfo
+    UserInstall(UserId),
+    Unknown(InstallationContext),
+}
+
+#[cfg_attr(feature = "typesize", derive(typesize::derive::TypeSize))]
+#[cfg(feature = "unstable_discord_api")]
+#[derive(Clone, Debug, Default)]
+#[repr(transparent)]
+pub struct AuthorizingIntegrationOwners(Vec<AuthorizingIntegrationOwner>);
+
+#[cfg(feature = "unstable_discord_api")]
+impl<'de> serde::Deserialize<'de> for AuthorizingIntegrationOwners {
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> StdResult<Self, D::Error> {
+        struct Visitor;
+
+        impl<'de> serde::de::Visitor<'de> for Visitor {
+            type Value = AuthorizingIntegrationOwners;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                formatter.write_str("a hashmap containing keys of InstallationContext and values based on those keys")
+            }
+
+            fn visit_map<A>(self, mut map: A) -> StdResult<Self::Value, A::Error>
+            where
+                A: serde::de::MapAccess<'de>,
+            {
+                let mut out = Vec::new();
+                while let Some(key_str) = map.next_key::<serde_cow::CowStr<'_>>()? {
+                    let key_int = key_str.0.parse::<u8>().map_err(serde::de::Error::custom)?;
+                    let value = match InstallationContext::from(key_int) {
+                        InstallationContext::Guild => {
+                            AuthorizingIntegrationOwner::GuildInstall(map.next_value()?)
```

Comment:
alternatively of course
```suggestion
                            AuthorizingIntegrationOwner::GuildInstall(map.next_value().ok())
```